### DISCUSSION TITLE
Issues with creating the .app bundle and the Info.plist

### DISCUSSION THREAD

**Discussion by Irasil** [AvaloniaUI/Avalonia#13884] [Q&A]
I created a test project ([here](https://github.com/Irasil/AvaloniaPListTest/tree/main)) to test the creation of the .app bundle and signing. For a while, it was working, and I could override the app name using the `CFBundleName` key. However, it's not working anymore. The only thing I changed was updating Visual Studio for Mac to 17.6.7. But even when I try to create the .app using Publish under Windows and then create a bundle with the Avalonia script, the Info.plist is ignored.

The command I use to publish it: `dotnet publish -r osx-x64 --configuration Release -p:UseAppHost=true --self-contained /p:CreatePackage=false`

Here is my script:

```bash
#!/bin/bash
APP_NAME="/Users/simon/Desktop/test/IBI-aws.app"
PUBLISH_OUTPUT_DIRECTORY="/Users/simon/Plist/Plist.Desktop/bin/Release/net7.0-macos/osx-x64/."
# PUBLISH_OUTPUT_DIRECTORY should point to the output directory of your dotnet publish command.
# One example is /path/to/your/csproj/bin/Release/netcoreapp3.1/osx-x64/publish/.
# If you want to change output directories, add `--output /my/directory/path` to your `dotnet publish` command.
INFO_PLIST="Info.plist"
ICON_FILE="car.icns"

if [ -d "$APP_NAME" ]
then
    rm -rf "$APP_NAME"
fi

mkdir "$APP_NAME"

mkdir "$APP_NAME/Contents"
mkdir "$APP_NAME/Contents/MacOS"
mkdir "$APP_NAME/Contents/Resources"

cp "$INFO_PLIST" "$APP_NAME/Contents/Info.plist"
cp "$ICON_FILE" "$APP_NAME/Contents/Resources/$ICON_FILE"
cp -a "$PUBLISH_OUTPUT_DIRECTORY" "$APP_NAME/Contents/MacOS"

```

And here is my Info.plist.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleIconFile</key>
	<string>car.icns</string>
	<key>CFBundleIdentifier</key>
	<string>com.identifier</string>
	<key>CFBundleName</key>
	<string>MyApp</string>
	<key>CFBundleVersion</key>
	<string>1.0.0</string>
	<key>LSMinimumSystemVersion</key>
	<string>10.12</string>
	<key>CFBundleExecutable</key>
	<string>Plist.Desktop</string>
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>CFBundleShortVersionString</key>
	<string>1.0</string>
	<key>CFBundleAllowMixedLocalizations</key>
	<false/>
	<key>CFBundleDisplayName</key>
	<string>MyApp</string>
	<key>NSHighResolutionCapable</key>
	<true/>
</dict>
</plist>
```

Does anyone know what exactly I'm doing wrong or why this isn't working as desired?

**Comment by thevortexcloud** [comment]
Apparently it can be ignored if `CFBundleDisplayName` is set instead. If that's being set somewhere, I am not sure. I am not seeing it in Avalonia's code though (but it may be getting set some other way). It does get used in a few of the samples however. So I guess try that instead?

**Reply by thevortexcloud** [reply]
Avalonia apparently also uses the name defined in `App.axaml` on the application node for stuff like this. 

https://github.com/AvaloniaUI/Avalonia/blob/0a6987b0855bc7cd0b371571bb0234dc779da181/src/Avalonia.Controls/Application.cs#L307-L314

So that may also work. Although I am not clear how it gets used as it seemingly does nothing for me. But that just might be a platform problem.

**Reply by Irasil** [reply]
Sorry, I didn't express myself very clearly. I wanted to test the `Info.plist` because I assumed it would be important for deployment. To test if it works with the `Info.plist`, I chose the simplest key (`CFBundleDisplayName`) to test.

According to [this](https://github.com/AvaloniaUI/Avalonia/issues/3541#issuecomment-1131857633) comment, the `.plist` should override the app's name. In the mentioned project above, I set the name to "Test" in the `app.xaml`, and in the `.plist`, I set `CFBundleName` and `CFBundleDisplayName` to "MyApp". I described in [this](https://github.com/AvaloniaUI/Avalonia/discussions/13568#discussioncomment-7575788) comment that it worked.

Unfortunately, it's not working anymore, and I absolutely don't understand why? At the moment this problem is rather minor. If I can't solve the window title issue, I don't have to solve this problem :)
